lib/matplotlib/tests/test_category.py

```diff
@@ -63,27 +67,34 @@ class TestStrCategoryConverter(object):
     ref: /pandas/tseries/tests/test_converter.py
          /pandas/tests/test_algos.py:TestFactorize
     """
-    testdata = [("Здравствуйте мир", [("Здравствуйте мир", 42)], 42),
-                ("hello world", [("hello world", 42)], 42),
-                (['a', 'b', 'b', 'a', 'a', 'c', 'c', 'c'],
-                 [('a', 0), ('b', 1), ('c', 2)],
-                 [0, 1, 1, 0, 0, 2, 2, 2]),
-                (['A', 'A', np.nan, 'B', -np.inf, 3.14, np.inf],
-                 [('nan', -1), ('3.14', 0), ('A', 1), ('B', 2),
-                  ('-inf', 100), ('inf', 200)],
-                 [1, 1, -1, 2, 100, 0, 200])]
-    ids = ["unicode", "single", "basic", "mixed"]
+
+    test_cases = {"unicode": ("Здравствуйте мир",
```

Comment:
This needs to be an ordered dict so that they are always in the same order so that  pytest-xdist works.

Reply:
Kinda surprised since I was always taught that tests should be independent of each other/order but ok.

Reply:
they are, but xdist imports the tests in both processes and then sorts out who does what (instead of trying to pickle and ship tests across the wire).  One of it's start up checks is that it finds exactly the same tests in both processes (to make sure nothing funny is happening).  In 3.5 the order is random which is fine from a testing point of view but not fine from xdist counting it's fingers and toes point of view.